src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -21,77 +20,961 @@ const ActorInit En_Goma_InitVars = {
+            } else {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_LAND2);
+            }
+        }
+    }
+    this->unk_2C4 = 0;
+}
+
+void func_80A4A2EC(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 0.5f, 2.0f);
+    Math_SmoothScaleMaxS(&this->actor.shape.rot.y, func_8002DA78(&this->actor, &PLAYER->actor), 2, 0xBB8);
+    if (this->unk_2CC == 0) {
+        func_80A49F10(this);
+    }
+}
+
+void func_80A4A368(EnGoma* this, GlobalContext* globalCtx) {
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_800A56C8(&this->skelAnime, 1.0f) || func_800A56C8(&this->skelAnime, 5.0f)) {
+        if (this->actor.params < 6) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_WALK);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_WALK);
+        }
+    }
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, 3.3333333f, 0.5f, 2.0f);
+    Math_SmoothScaleMaxS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 3, 0x7D0);
+    Math_SmoothScaleMaxS(&this->actor.shape.rot.y, this->actor.posRot.rot.y, 2, 0xBB8);
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.velocity.y = 0.0f;
+    }
+    if (this->actor.xzDistFromLink <= 150.0f) {
+        func_80A49F94(this);
+    }
+}
+
+void func_80A4A50C(EnGoma*, GlobalContext*);
+
+void func_80A4A470(EnGoma* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_80A4A50C;
+    this->unk_2F8 = 100;
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_06001548, -5.0f);
+    this->unk_2CC = (s16)Math_Rand_ZeroFloat(15.0f) + 3;
+    if (this->actor.params < 6) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_FREEZE);
+        return;
+    }
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+}
+
+void func_80A4A50C(EnGoma* this, GlobalContext* globalCtx) {
+    func_8003426C(&this->actor, 0, 0xB4, 0, 2);
+    this->unk_2C4 = 2;
+    if (this->unk_2CC) {
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    }
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.velocity.y = 0.0f;
+        Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 0.5f, 2.0f);
+    }
+    if (this->unk_2F8 == 0) {
+        func_80A49E80(this);
+        return;
+    }
+    this->unk_2F8--;
+    if (this->unk_2F8 < 30) {
+        if (this->unk_2F8 & 1) {
+            this->actor.posRot.pos.x += 1.5f;
+            this->actor.posRot.pos.z += 1.5f;
+        } else {
+            this->actor.posRot.pos.x -= 1.5f;
+            this->actor.posRot.pos.z -= 1.5f;
+        }
+    }
+}
+
+void func_80A4A608(EnGoma* this, GlobalContext* globalCtx) {
+    s16 tmp2;
+    s16 tmp;
+
+    tmp = func_8002DA78(&this->actor, &PLAYER->actor) - this->actor.shape.rot.y;
+    tmp2 = func_8002DAE0(&this->actor, &PLAYER->actor) - this->actor.shape.rot.x;
+
+    if (tmp >= 0x1771) {
```

Comment:
does the CLAMP macro work here?

Reply:
No unfortunately. No unconditional branches here, which the macro introduces with a ternary. Doesn't match.